x/escrow/handler_test.go

```diff
@@ -617,28 +617,28 @@ func TestHandler(t *testing.T) {
 
 			// set initial data
 			acct, err := cash.WalletWith(tc.account, tc.balance...)
-			require.NoError(t, err)
+			assert.Nil(t, err)
 			err = bank.Save(db, acct)
-			require.NoError(t, err)
+			assert.Nil(t, err)
 
 			// do delivertx
 			for j, p := range tc.prep {
 				// try check
 				cache := db.CacheWrap()
 				_, err = router.Check(p.ctx(), cache, p.tx())
-				require.NoError(t, err, "%d", j)
+				assert.Nil(t, err)
 				cache.Discard()
 
 				// then perform
 				_, err = router.Deliver(p.ctx(), db, p.tx())
-				require.NoError(t, err, "%d", j)
+				assert.Nil(t, err)
 			}
 
 			_, err = router.Deliver(tc.do.ctx(), db, tc.do.tx())
 			if tc.isError {
-				require.Error(t, err)
+				assert.Nil(t, err)
```

Comment:
Isn't this the opposite of the intended test?

```go
if ! tc.isError {
  assert.Nil(t, err)
} else if err == nil {
  t.Fatal("Expected error but got nil")
}
```

At least this is the format I ended up with after reading plenty of Piotr's usage of testing.

Reply:
This testing logic was created before `Is` testing of errors existed. If you use `*errors.Error` as the tested value, you can [use `nil` value](https://github.com/iov-one/weave/blob/master/x/currency/model_test.go#L23) for testing as well :rocket: This way, you can ensure both that an error is of certain type or that no error was returned.

Reply:
yeah, this mistake squeezed in, good catch!